src/back/interfaces/router.py

```diff
@@ -0,0 +1,10 @@
+from typing import Self, Protocol
+from tornado.iostream import IOStream
+
+
+class MessageHandlerInterface(Protocol):
+    """Interface for handling messages in the router."""
+
+    def handle(self: Self, message: bytes, stream: IOStream) -> None:
```

Comment:
Since you are handling messages, the `message` type should be `MessageInterface`. `MessageReader` & `MessageWriter` are already responsible for serializing/deserializing messages

Reply:
Also make `stream` a `WritableStreamInterface`, so message handlers won't try to read from the stream